tools/cli/adminElasticSearchCommands.go

```diff
@@ -231,3 +256,251 @@ func generateESDoc(msg *indexer.Message) map[string]interface{} {
 	}
 	return doc
 }
+
+// This function is used to trim unnecessary tag in returned json for table header
+func bucketKey(k string) string {
+	// group key is in form of "group_key", we only need "key" as the column name
+	if strings.HasPrefix(k, "group_") {
+		k = k[6:]
+	}
+	if strings.HasPrefix(k, "Attr_") {
+		k = k[5:]
+	}
+	return fmt.Sprintf(`%v(*)`, k)
+}
+
+// parse the returned time to readable string if time is in int64 format
+func toTimeStr(s interface{}) string {
+	floatTime, err := strconv.ParseFloat(s.(string), 64)
+	intTime := int64(floatTime)
+	if err != nil {
+		return s.(string)
+	}
+	t := time.Unix(0, intTime)
+	return t.Format(time.RFC3339)
+}
+
+// GenerateReport generate report for an aggregation query to ES
+func GenerateReport(c *cli.Context) {
+	// use url command argument to create client
+	url := getRequiredOption(c, FlagURL)
+	index := getRequiredOption(c, FlagIndex)
+	sql := getRequiredOption(c, FlagListQuery)
+	reportFormat := getRequiredOption(c, FlagOutputFormat)
```

Comment:
Let's don't make it required. 
Default is no output, only show in CLI STDOUT
if provided, them use html/csv

use c.IsSet() and c.String()